website/docs/r/database_secret_backend_connection.md

```diff
@@ -0,0 +1,183 @@
+---
+layout: "vault"
+page_title: "Vault: vault_database_secret_backend_connection resource"
+sidebar_current: "docs-vault-resource-database-secret-backend-connection"
+description: |-
+  Configures a database secret backend connection for Vault.
+---
+
+# vault\_database\_secret\_backend\_connection
+
+Creates a Database Secret Backend connection in Vault. Database secret backend
+connections can be used to generate dynamic credentials for the database.
+
+~> **Important** All data provided in the resource configuration will be
+written in cleartext to state and plan files generated by Terraform, and
+will appear in the console output when Terraform runs. Protect these
+artifacts accordingly. See
+[the main provider documentation](../index.html)
+for more details.
+
+## Example Usage
+
+```hcl
+resource "vault_mount" "db" {
+  path = "postgres"
+  type = "database"
+}
+
+resource "vault_database_secret_backend_connection" "postgres" {
+  backend       = "${vault_mount.db.path}"
+  name          = "postgres"
+  allowed_roles = ["dev", "prod"]
+
+  postgresql {
+    connection_url = "postgres://username:password@host:port/database"
+  }
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `name` - (Required) A unique name to give the database connection.
+
+* `backend` - (Required) The unique name of the Vault mount to configure.
+
+* `verify_connection` - (Optional) Whether the connection should be verified on
+  initial configuration or not.
+
+* `allowed_roles` - (Optional) A list of roles that are allowed to use this
+  connection.
+
+* `cassandra` - (Optional) Configuration options for Cassandra connections.
```

Comment:
For these, I think it would help to explicitly say "A nested block containing ..." to help explain how the attributes described in the subsequent sections are to be used, since this is not usual idiom so it may not be immediately obvious what we're expecting here.

Perhaps also a paragraph at the end of this list explaining that at exactly one of these must be provided and the chosen block decides which database plugin is used... I'm thinking about someone who is familiar with Vault trying to map their understanding onto these docs, where we've kinda hidden the idea of a "plugin" and so we don't have that hook here to help someone figure out how that concept maps.